test/config_parser_SUITE.erl

```diff
@@ -2910,64 +2921,33 @@ service_mongoose_system_metrics(_Config) ->
     ?err(T(#{<<"tracking_id">> => #{<<"secret">> => 666, <<"id">> => 666}})),
     ?err(T(#{<<"report">> => <<"maybe">>})).
 
+%% Logs
+
 no_warning_about_subdomain_patterns(_Config) ->
     check_module_defaults(mod_vcard),
     check_iqdisc(mod_vcard),
     P = [modules, mod_vcard],
     T = fun(Opts) -> #{<<"modules">> => #{<<"mod_vcard">> => Opts}} end,
-
-    Node = #{node => mongooseim@localhost},
-    logger_ct_backend:start(Node),
-    logger_ct_backend:capture(warning, Node),
-
     ?cfgh(P ++ [host], {prefix, <<"vjud.">>},
           T(#{<<"host">> => <<"vjud.@HOST@">>})),
+    ?assertNoLog(warning, #{what := cfg_validate_domain}),
+
     ?cfgh(P ++ [host], {fqdn, <<"vjud.test">>},
           T(#{<<"host">> => <<"vjud.test">>})),
-
-    logger_ct_backend:stop_capture(Node),
-    logger_ct_backend:stop(Node),
-
-    FilterFun = fun(_, Msg) ->
-                        re:run(Msg, "test") /= nomatch orelse
```

Comment:
Just getting rid of these regexes and actually doing struct matching is such a great win!